pkg/handlers/profiling_handlers.go

```diff
@@ -0,0 +1,97 @@
+package handlers
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/openshift/node-observability-agent/pkg/runs"
+)
+
+const (
+	defaultCrioHost    = "localhost"
+	defaultCrioPort    = "6060"
+	defaultKubeletPort = "10250"
+	crioProfilePath    = "debug/pprof/profile"
+	kubeletProfilePath = "debug/pprof/profile"
+)
+
+// profileCrio triggers CRIO profiling on localhost.
+func (h *Handlers) profileCrio(uid string, client *http.Client) runs.ProfilingRun {
+	u := url.URL{
+		Scheme: "http",
+		Host:   net.JoinHostPort(defaultCrioHost, defaultCrioPort),
+		Path:   crioProfilePath,
+	}
+	hlog.Infof("Requesting CRIO profiling")
+	r := sendHTTPProfileRequest(runs.CrioRun, "GET", u.String(), "", h.crioPprofOutputFilePath(uid), client)
+	if r.Successful {
+		hlog.Info("CRIO profiling request successfully finished")
+	}
+
+	return r
+}
+
+// profileKubelet triggers Kubelet profiling on h.NodeIP using h.Token for authorization.
+func (h *Handlers) profileKubelet(uid string, client *http.Client) runs.ProfilingRun {
+	u := url.URL{
+		Scheme: "https",
+		Host:   net.JoinHostPort(h.NodeIP, defaultKubeletPort),
+		Path:   kubeletProfilePath,
+	}
+	hlog.Infof("Requesting Kubelet profiling")
+	r := sendHTTPProfileRequest(runs.KubeletRun, "GET", u.String(), h.Token, h.kubeletPprofOutputFilePath(uid), client)
+
+	if r.Successful {
+		hlog.Info("Kubelet profiling request successfully finished")
+	}
+
+	return r
+}
+
+// sendHTTPProfileRequest sends the http request to the given url,
+// writes the response down to the given output and returns the profiling run instance.
+func sendHTTPProfileRequest(rtype runs.RunType, method, url, token, outputPath string, client *http.Client) runs.ProfilingRun {
+	run := runs.ProfilingRun{
```

Comment:
I think we should have some way to distinguish runs (type and id) for the error logs.